Build-Task.ps1

```diff
@@ -0,0 +1,244 @@
+<#
+    .SYNOPSIS
+    Install common task modules at build time.
+
+    .DESCRIPTION
+    Install common task modules at build time.
+
+    .PARAMETER TaskRoot
+    The root of the task to be built
+
+    .PARAMETER Clean
+    Clean package directories before building
+
+    .PARAMETER Build
+    Package an extension package
+
+    .PARAMETER Publish
+    Packge and publish an extension package to the marketplace
+
+    .PARAMETER AccessToken
+    A personal access token with enough privileges to publish the package
+
+    .PARAMETER SkipRestore
+    Don't restore dependencies listed in dependency.json
+
+    .EXAMPLE
+    Restore-TaskDependency -TaskRoot ./MyTask
+
+    .EXAMPLE
+    Restore-TaskDependency -TaskRoot ./MyTask -Clean
+
+    .EXAMPLE
+    Restore-TaskDependency -TaskRoot ./MyTask -Clean -Build
+
+    .EXAMPLE
+    Restore-TaskDependency -TaskRoot ./MyTask -Clean -Publish
+
+    .EXAMPLE
+    Restore-TaskDependency -TaskRoot ./MyTask -Publish -SkipRestore
+
+    .NOTES
+    Requirements:
+    The following applications must be installed an available in your PATH
+    - git
+    - tfx-cli
+#>
+[CmdletBinding(DefaultParameterSetName = "Build")]
+Param(
+    [Parameter(Mandatory = $true)]
+    [System.Io.FileInfo]$TaskRoot,
+    [Parameter(Mandatory = $false)]
+    [switch]$Clean,
+    [Parameter(Mandatory = $false)]
+    [Parameter(Mandatory = $true, ParameterSetName = "Build")]
+    [switch]$Build,
+    [Parameter(Mandatory = $false, ParameterSetName = "Publish")]
+    [switch]$Publish,
+    [Parameter(Mandatory = $true, ParameterSetName = "Publish")]
+    [string]$AccessToken,
+    [Parameter(Mandatory = $false)]
+    [switch]$SkipRestore
+)
+
+# Default error action to stop
+$ErrorActionPreference = "Stop"
```

Comment:
Are you happy with using a global error action?

Reply:
i'll look to remove this